pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -40,7 +40,7 @@ def work(self):
 
         if self.evolve_all[0] != 'all':
             # check for negation
-            negate = filter(lambda x: x[0] == '-', self.evolve_all)
+            negate = filter(lambda x: len(x) > 0 and x[0] == '-', self.evolve_all)
```

Comment:
Could we have a check when we check for it being 'none' as well (though this might cover both).